polygeist/test/polygeist-opt/sycl/argpromotion.mlir

```diff
@@ -49,7 +49,7 @@ gpu.module @device_func {
   // COM: This function is a candidate, check that it is transformed correctly.
   func.func private @callee3(%arg0: memref<?x!llvm.struct<(i32)>>, %arg1: memref<?x!llvm.struct<(f32)>>) {
```

Comment:
If the original callee with arguments `%arg0: memref<?x!llvm.struct<(i32)>>, %arg1: memref<?x!llvm.struct<(i32)>>`, 
`%arg0` and `%arg1` could be alias, then is it unsafe to put `noalias` on the new arguments `%arg0: memref<?xi32>, %arg1: memref<?xi32>`?

Reply:
Changed the code so that the noalias attribute is added only if we peel a single argument (there are more complex schemes that are possible but this is sufficient for our use case).